fix(search): execution results < limit == no more matches

emjburns opened the pull request:
A user discovered a situation where SQL behaves differently than Redis:
```
$ SIZE=100
$ curl -s "https://spinnaker/applications/my_app/executions/search?application=my_app&pipelineName=DEPLOY&triggerTimeStartBoundary=1539846060000&triggerTimeEndBoundary=1539964800000&reverse=true&size=$SIZE" | jq '. | length'
27

$ SIZE=1000
$ curl -s "https://spinnaker/applications/my_app/executions/search?application=my_app&pipelineName=DEPLOY&triggerTimeStartBoundary=1539846060000&triggerTimeEndBoundary=1539964800000&reverse=true&size=$SIZE" | jq '. | length'
98
```

This happens because with redis, we'd get all the executions and then filter down to the matching ones. In sql, we are only getting `limit` number of executions and then filtering them down to the matching ones. This is unexpected.

This PR does a couple of things:
 - removes `Observables` from this endpoint, since we immediately block.
 - creates a new execution repository endpoint which returns all executions in a time period, sorted how you specify. This allows each executionRepositoryImpl to handle pagination and sorting however it would like.
- fixed the problem stated above.

robzienert commented on orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/pipeline/persistence/SqlExecutionRepository.kt:
```diff
@@ -405,24 +405,69 @@ class SqlExecutionRepository(
+        page,
+        limit
+      )
+      moreResults = results.size == limit
```
I'm always partial to a nice `results.size >= limit` myself, because I don't trust computers or myself.

emjburns replied:
Changed!

robzienert commented on orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy:
```diff
@@ -312,52 +313,63 @@ class TaskController {
+    ExecutionCriteria executionCriteria =  new ExecutionCriteria()
+      .setSortType( reverse ? REVERSE_BUILD_TIME : BUILD_TIME )
+    if (statuses != null && statuses != "") {
+      executionCriteria.setStatuses(statuses.split(",").toList())
```
Hmm. I wonder if it'd be useful to have a `ExecutionCriteria setStatuses(String statuses)` that will use a splitter?

emjburns replied:
It's easy enough to do at the call site, I don't think adding a method would provide a lot of value.

emjburns commented:
@robzienert previously sorting was done in the controller, and I've moved it do be done in the execution repository. That's why I changed the sort order. I _think_ it still works the same 🤞 based on my tests.
